server/src/helpers/passport.config.ts

```diff
@@ -1,50 +1,73 @@
 import passport from 'passport';
 import {Strategy as LocalStrategy} from 'passport-local';
+import {OAuth2Strategy as GoogleStrategy} from 'passport-google-oauth';
 import bcrypt from 'bcrypt';
 
 import {IUserModel, User} from '../models/user.model';
+import {createGoogleUser} from './google.auth';
+import {checkUserByProp} from '../common.db.request/user.checkEmail';
 
 passport.use(
-    'register',
-    new LocalStrategy(
-        async (username: string, password: string, done: any): Promise<void> => {
-            try {
-                const user = await User.findOne({username});
-                return done(null, user);
-            } catch (e) {
-                return done(e);
-            }
-        },
-    ),
+  'register',
+  new LocalStrategy(
+    async (username: string, password: string, done: any): Promise<void> => {
+      try {
+        const user = await checkUserByProp(username);
+        return done(null, user);
+      } catch (e) {
+        return done(e);
+      }
+    },
+  ),
 );
 
 passport.use(
-    'local',
-    new LocalStrategy({
-        usernameField: 'email',
-        passwordField: 'password',
-    }, (username: string, password: string, done: any): void => {
-        User.findOne({email: username}, (err: Error, user: IUserModel) => {
-            if (err) {
-                return done(err);
-            }
-            if (!user) {
-                return done(null, false);
-            }
+  'local',
+  new LocalStrategy({
+    usernameField: 'email',
+    passwordField: 'password',
+  }, async (email: string, password: string, done: any): Promise<void> => {
 
-            bcrypt.compare(password, user.password, (error: Error, result: boolean) => {
-                if (result) {
-                    return done(null, user);
-                }
-                return done(err, false, {message: 'Incorrect'});
-            });
-        });
-    }));
+    try {
+      const user = await checkUserByProp(email);
+      if (!user) {
+        return done(null, false);
+      }
+      bcrypt.compare(password, user.password, (error: Error, result: boolean) => {
+        if (result) {
+          return done(null, user);
+        }
+      });
+    } catch (e) {
+      return done(e);
+    }
+  }),
+);
+
+passport.use(new GoogleStrategy({
+    clientID: process.env.GOOGLE_CLIENT_ID!,
+    clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
+    callbackURL: process.env.GOOGLE_CALLBACK_URL!,
```

Comment:
you can create common or config directory with config file with all needed common variable
export const GOOGLE_CLIENT_ID= process.env.GOOGLE_CLIENT_ID;

etc

and import here 

if you will have a type error, try 
```
const GOOGLE_CLIENT_ID: string = (process.env.GOOGLE_CLIENT_ID as string);
```

or some another solution